File: frontend/between_pages/lib/models/catalog/manga_response_dto.dart
class MangaResponseDTO {
  final int? idManga;
  final String? mangadexId;
  final String? source;
  final String? title;
  final String? mangaka;
  final String? demographic;
  final String? genre;
  final String? description;
  final String? coverUrl;
  final int? totalChapters;
  final int? totalVolumes;
  final String? publicationStatus;

  MangaResponseDTO({
    this.idManga,
    this.mangadexId,
    this.source,
    this.title,
    this.mangaka,
    this.demographic,
    this.genre,
    this.description,
    this.coverUrl,
    this.totalChapters,
    this.totalVolumes,
    this.publicationStatus,
  });

  factory MangaResponseDTO.fromJson(Map<String, dynamic> json) {
    return MangaResponseDTO(
      idManga: int.tryParse(json['id']?.toString() ?? '0'),
      mangadexId: json['mangadex_id'] as String?,
      source: json['source'] as String?,
      title: json['title'] as String?,
      mangaka: json['author'] as String?,
      demographic: json['demographic'] as String?,
      genre: json['genre'] as String?,
      description: json['description'] as String?,
      coverUrl: json['cover_url'] as String?,
      totalChapters: json['total_chapters'] as int?,
      totalVolumes: json['total_volumes'] as int?,
      publicationStatus: json['publication_status'] as String?,
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': idManga,
      'mangadex_id': mangadexId,
      'source': source,
      'title': title,
      'author': mangaka,
      'demographic': demographic,
      'genre': genre,
      'description': description,
      'cover_url': coverUrl,
      'total_chapters': totalChapters,
      'total_volumes': totalVolumes,
      'publication_status': publicationStatus,
    };
  }
}

File: backend/src/main/java/com/calonuria/backend/dto/catalog/MangaResponseDTO.java
package com.calonuria.backend.dto.catalog;

import com.fasterxml.jackson.annotation.JsonProperty;
import lombok.Data;

/**
 * DTO para la respuesta con información de manga.
 */
@Data
public class MangaResponseDTO {

    private Long id;

    @JsonProperty("mangadex_id")
    private String mangadexId;

    private String source;
    private String title;
    private String author;
    private String demographic;
    private String genre;
    private String description;

    @JsonProperty("cover_url")
    private String coverUrl;

    @JsonProperty("total_chapters")
    private Integer totalChapters;

    @JsonProperty("total_volumes")
    private Integer totalVolumes;

    @JsonProperty("publication_status")
    private String publicationStatus;
}